File: solutions/Python/1768.MergeStringsAlternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        res: string = ""
        i: int = 0
        lenWord1: int = len(word1)
        lenWord2: int = len(word2)

        while (i < lenWord1 or i < lenWord2):
            if i < lenWord1:
                res += word1[i]
            if i < lenWord2:
                res += word2[i]
            i += 1

        return res

File: solutions/TypeScript/1768.MergeStringsAlternately.ts
function mergeAlternately(word1: string, word2: string): string {
  let res: string = "";
  let i: number = 0;
  const word1Length: number = word1.length;
  const word2Length: number = word2.length;

  while (i < Math.max(word1Length, word2Length)) {
    if (i < word1Length) res += word1[i];
    if (i < word2Length) res += word2[i];
    i++;
  }

  return res;
}
